backtracking/remove_invalid_parenthese.py

```diff
@@ -0,0 +1,120 @@
+
+    Args:
+        input_string (str): The input string containing parentheses.
+
+    Returns:
+        List[str]: A list of valid strings after removing
+        the minimum number of parentheses.
+
+    Examples:
+        >>> remove_invalid_parentheses(")(a)())")
+        ['(a())', '(a)()']
+
+        >>> remove_invalid_parentheses("()())()")
+        ['(())()', '()()()']
+
+        >>> remove_invalid_parentheses(")(")
+        ['']
+    """
+
+    def is_valid(next_string: str) -> bool:
+        """
+        Helper function to check if a string has valid parentheses.
+
+        Params:
+            next_string (str): The input string to be checked.
+        Returns:
+            bool: True if the parentheses in the string are valid, False otherwise.
+
+        Examples:
+        >>> is_valid("(a)())")
+        False
+
+        >>> is_valid("(a())")
+        True
+
+        >>> is_valid(")(a)()")
+        False
+
+        """
+        count: int = 0
+        for char in next_string:
+            if char == "(":
+                count += 1
+            elif char == ")":
+                count -= 1
+                if count < 0:
+                    return False
+        return count == 0
+
+    def dfs(
+        current_string: str, start_index: int, left_removed: int, right_removed: int
+    ) -> None:
+        """
+        Depth-first search function to generate valid combinations
+        of the input string.
+
+        Params:
+            current_string (str): The current string being processed.
+            start_index (int): The index to start processing from.
+            left_removed (int): The number of left parentheses to be removed.
+            right_removed (int): The number of right parentheses to be removed.
+        Returns:
+            None
+        Examples:
+        >>> dfs("()())()", 0, 0, 1)
+        None
+
+        >>> dfs("()())(", 6, 0, 0)
+        None
+
+        >>> dfs(")(a)())", 0, 0,, 2)
+        None
+        """
+        if left_removed == 0 and right_removed == 0:
+            if is_valid(current_string):
+                valid_parentheses.add(current_string)
+            return
+        for i in range(start_index, len(current_string)):
+            if i > start_index and current_string[i] == current_string[i - 1]:
+                continue
+            if current_string[i] == "(" and left_removed > 0:
+                dfs(
+                    current_string[:i] + current_string[i + 1 :],
+                    i,
+                    left_removed - 1,
+                    right_removed,
+                )
+            elif current_string[i] == ")" and right_removed > 0:
+                dfs(
+                    current_string[:i] + current_string[i + 1 :],
+                    i,
+                    left_removed,
+                    right_removed - 1,
+                )
+
+    valid_parentheses: set = set()
+    left_removed_count: int = 0
+    right_removed_count: int = 0
```

Comment:
```suggestion
    valid_parentheses = set()
    left_removed_count = 0
    right_removed_count = 0
```
Python is dynamically typed, so these your annotations are infered

Reply:
i know the pyhton is dynamically typed, but the pre-commit.ci - pr needs it.

otherwisw its will give error: Need type annotation for "valid_parentheses" (hint: "valid_parentheses: Set[<type>] = ...")